sway/input/input-manager.c

```diff
@@ -659,8 +705,16 @@ void input_manager_set_focus(struct sway_node *node) {
 void input_manager_apply_input_config(struct input_config *input_config) {
 	struct sway_input_device *input_device = NULL;
 	bool wildcard = strcmp(input_config->identifier, "*") == 0;
+	bool type_wildcard = strncmp(input_config->identifier, "type:", 5) == 0;
 	wl_list_for_each(input_device, &server.input->devices, link) {
+		if (type_wildcard &&
```

Comment:
This should occur in `store_input_config`, not here. Merging it here will make it so any device specific options will always be shadowed by the type config.

Reply:
Sorry, this process has been really long because I haven't been really sure about how the layering of the configs should be handled. I want to get it right this time!

So to clarify, when a type config is stored, we merge it on top of the existing device configs. When a new device appears, we use the device config, merged on top of the type config.

Reply:
There is no rush. We would rather have things done right than fast. There are no deadlines or anything to meet. We are all volunteering our time after all.

That sounds correct. You'll either have to search for a device that matches the config or store the type in the input config when a device matches it.

On a related note, I just realized that `merge_wildcard_on_all` needs to be modified to merge on top of the type configs as well as the device configs

Reply:
> There is no rush. We would rather have things done right than fast. There are no deadlines or anything to meet. We are all volunteering our time after all.

Wasn't implying there was a rush, I just felt like I was wasting your time by doing this with trial and error basically. That being said, I'm glad you guys actually take time to review things thoroughly and it makes me a lot more appreciative of the work you guys do, and me more happy to use it.